src/m1_facilitator/services/ConfirmDepositService.ts

```diff
@@ -0,0 +1,167 @@
+    messageRepository: MessageRepository,
+    depositIntentRepository: DepositIntentRepository,
+    auxiliaryTransactionExecutor: TransactionExecutor,
+  ) {
+    super();
+    this.originWeb3 = originWeb3;
+    this.auxiliaryWeb3 = auxiliaryWeb3;
+    this.messageRepository = messageRepository;
+    this.depositIntentRepository = depositIntentRepository;
+    this.auxiliaryTransactionExecutor = auxiliaryTransactionExecutor;
+  }
+
+  /**
+   * Triggers on update in gateway model.This service checks for pending
+   * messages based on below condition:
+   *  1. Message Type should be Deposit
+   *  2. Source declaration block height of message should be less than equals
+   *     to remote gateway last proven block height.
+   *  3. Target status of message should be undeclared.
+   *  4. Message should be initiated from given gateway.
+   *
+   * If there are pending messages based on above criteria, confirm deposit
+   * transaction is sent.
+   *
+   * @param gateways List of gateway object.
+   */
+  public async update(gateways: Gateway[]) {
+    const confirmMessagePromises = gateways.map(async (gateway): Promise<void> => {
+      const messages = await this.messageRepository.getPendingMessagesByGateway(
+        gateway.gatewayGA,
+        MessageType.Deposit,
+        gateway.remoteGatewayLastProvenBlockNumber,
+      );
+
+      if (messages.length > 0) {
+        await this.confirmMessages(messages, gateway);
+      }
+    });
+
+    await Promise.all(confirmMessagePromises);
+  }
+
+  /**
+   * Creates and send confirm deposit transaction.
+   *
+   * @param messages List of messages which needs to be confirmed.
+   * @param gateway Instance of gateway contract.
+   */
+  private async confirmMessages(messages: Message[], gateway: Gateway): Promise<void> {
+    const confirmMessageTransactionPromises = messages.map(async (message): Promise<void> => {
+      const depositIntent = await this.depositIntentRepository.get(message.messageHash);
+      if (depositIntent !== null) {
+        const rawTransaction = await this.confirmDepositIntentTransaction(
+          message,
+          depositIntent,
+          gateway,
+        );
+
+        await this.auxiliaryTransactionExecutor.add(rawTransaction);
+      }
+    });
+    await Promise.all(confirmMessageTransactionPromises);
+  }
+
+  /**
+   * Creates confirm deposit raw transaction.
+   *
+   * @param message Message Object.
+   * @param depositIntent Deposit intent object.
+   * @param gateway Gateway object.
+   */
+  private async confirmDepositIntentTransaction(
+    message: Message,
+    depositIntent: DepositIntent,
+    gateway: Gateway,
+  ): Promise<TransactionObject<string>> {
+    const cogatewayAddress = gateway.remoteGA;
+    const erc20Cogateway = Mosaic.interacts.getERC20Cogateway(this.auxiliaryWeb3, cogatewayAddress);
+
+    const blockNumber = gateway.remoteGatewayLastProvenBlockNumber;
+    const proof = await new ProofGenerator(this.originWeb3).getOutboxProof(
```

Comment:
pass the storage index.

Reply:
The proof generator may not work as per the expectations.
The path formation is different for `Mosaic-1` contracts

Reply:
Merging this, I will create a ticket to fix it in `@openst/mosaic-proof`